pybamm/expression_tree/concatenations.py

```diff
@@ -80,7 +81,8 @@ def evaluate(self, t=None, y=None):
 
 
 class DomainConcatenation(Concatenation):
-    """A node in the expression tree representing a concatenation of symbols.
+    """A node in the expression tree representing a concatenation of symbols, being
+    careful about domains.
```

Comment:
DomainConcatenation is only for combining variables on subdomains to make a new variable on a combined subdomain? (e.g. the electrolyte)